samples/cma_es.py

```diff
@@ -122,21 +122,22 @@ class ReturnTracker(Callback):
     def on_loop_start(self):
         self.ret = 0
 
-    def on_step_taken(self, transition, info):
+    def on_step_taken(self, step, transition, info):
```

Comment:
do czego potrzebujemy ten parametr w callbacku? gdzie to jest używane?

Reply:
To jest framework, to nie jest tutaj używane. Wzorowałem się na Kerasie, gdzie przekazywany jest numer epoki i numer batch-a. Widocznie ludzie (ja też) z tego korzystają. Tutaj przekazuję numer epizodu i numer kroku, potrafię sobie wyobrazić, że może to się przydać.

Reply:
Widocznie tam ktoś oczekiwał że to będzie dodane. Obecnie nie widzę potrzeby rozszerzania API jeżeli nie ma na to potrzeby. Jeżeli już mowa o rozszerzaniu API, szczególnie callbacków to wydaje mi się że dobrym pomysłem by było przekazywanie parametrów jako kwargs. Wtedy można przekazywać dodatkowy parametr bez zmiany wszystkich implementacji.

Reply:
Ogólnie zgadzam się, że nie ma sensu rozszerzać API bez potrzeby. To jest bardzo dobra i ważna zasada. Ale z drugiej strony to jest coś tak oczywistego i powszechnego we wszystkich frameworka, że nie trudno sobie wyobrazić, że znajdą się przypadki kiedy to będzie potrzebne.
Dodatkowo i tak robiłem refactor callbacków na potrzeby zbierania i zwracania tych "historii". Więc bardzo łatwo było zrobić to przy okazji i napewno dużo lepiej zrobić to teraz za jednym bólem (zmiana API wiąże się z potrzebą przerobienia wszystkich miejsc we wszystkich projektach które używają framework-a, na szczęście narazie mamy tylko dwa projekty, które są aktywne).

Jeśli chodzi o kwargs to dobry pomysł na przyszłość, choć raczej nie będzie to już potrzebne, bo można wszystko przekazać w `info` co kto tam potrzebuje.

Reply:
ok